src/src/Woo/ZipValidator.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace QIT_CLI\Woo;
+
+use ZipArchive;
+
+class ZipValidator {
+
+	/**
+	 * @var string[]
+	 */
+	protected $system_files = [
+		// https://github.com/github/gitignore/blob/main/Global/Windows.gitignore.
+		'Thumbs.db',
+		'Thumbs.db:encryptable',
+		'Desktop.ini',
+		'desktop.ini',
+		'ehthumbs.db',
+		'ehthumbs_vista.db',
+		'$RECYCLE.BIN/',
+		// https://github.com/github/gitignore/blob/main/Global/Linux.gitignore.
+		'~',
+		'.directory',
+		// https://github.com/github/gitignore/blob/main/Global/macOS.gitignore.
+		'.DS_Store',
+		'.AppleDouble',
+		'.LSOverride',
+		'.Spotlight-V100',
+		'.Trashes',
+		'.fseventsd',
+	];
+
+	/**
+	 * @param string $filepath
+	 *
+	 * @return \ZipArchive
+	 * @throws \RuntimeException If the file is not a valid zip file.
+	 */
+	private function open_file( string $filepath ) {
+		$zip    = new ZipArchive();
+		$opened = $zip->open( $filepath, ZipArchive::CHECKCONS );
+
+		if ( $opened === 21 ) {
+			throw new \RuntimeException( 'Zip file is inconsistent. Zip files generated by the Archive Utility from macOS may be the cause.' );
```

Comment:
A nitpick, but here we are failing any zips generated by MacOS Archive Utility.

What about instead, we do this?

If it fails the consistency check in a way that matches those zips generated by MacOS Archive Utility, we proceed with the script and show a warning to the user.

```php
	private function open_file( string $filepath ) {
		if ( $opened === 21 ) {
			throw new \UnexpectedValueException( 'Zip file is inconsistent. Zip files generated by the Archive Utility from macOS may be the cause.' );
		}
	}

	public function validate_zip( string $filepath ) {
		try {
			$zip = $this->open_file( $filepath );
		} catch( \UnexpectedValueException $e ) {
			// Let the user know if it fails consistency check because it was generated on Mac, but proceed.
			// Any other exception (such as RuntimeException) will not be caught and will interrupt the script.
			\QIT_CLI\App::make( \QIT_CLI\IO\Output::class )->writeln( sprintf( '<comment>%s</comment>', $e->getMessage() ) );
		}
	}
```

Reply:
Implemented in 6ba3ed8...060fa17

<img width="879" alt="Screenshot 2024-05-09 at 15 10 49" src="https://github.com/woocommerce/qit-cli/assets/532402/b5951ff7-1390-464f-a971-cb157228d4f5">